modules/rest-api/pages/rest-bucket-create.adoc

```diff
@@ -620,6 +626,114 @@ curl -v -X POST http://10.143.201.101:8091/pools/default/buckets/testBucket \
+The value can be either `true` (the default) or `false`.
+Note that the value is always `false` for the _&#95;default_ collection.
+
+This parameter is ignored unless the value of `storageBackend` is specified as `magma`.
+
+[#example-flushenabled-create]
+==== Example: Enable historyRetentionCollectionDefault, when Creating
+
+The following example creates a bucket, specifies its storage as _magma_, and specifies that a record of changes made to collections within the bucket should _not_ be made.
+
+----
+curl -X POST http://localhost:8091/pools/default/buckets \
+-u Administrator:password \
+-d name=testBucket \
+-d ramQuota=1100 \
+-d storageBackend=magma \
+-d historyRetentionCollectionDefault=false
+----
+
+Success returns `202 Accepted`.
+
+[#example-flushenabled-edit]
+==== Example: Modify historyRetentionCollectionDefault, when Editing
+
+The following example modifies the value of `historyRetentionCollectionDefault` for the existing bucket `testBucket`.
+
+----
+curl -v -X POST http://localhost:8091/pools/default/buckets/testBucket \
+-u Administrator:password \
+-d historyRetentionCollectionDefault=true
+----
+
+Success returns `200 OK`.
+
+[#historyretentionbytes]
+=== historyRetentionBytes
+
+Specifies the maximum size, in bytes, of the change history that is written to disk for each collection in the bucket (except the _&#95;default_ collection) when the value of `historyRetentionCollectionDefault` is `true`.
+
+[#example-flushenabled-create]
+==== Example: Set historyRetentionBytes, when Creating
+
+The following example creates a bucket, specifies its storage as _magma_, accepts the default value of `true` for `historyRetentionCollectionDefault`, and specifies the maximum disk-size of the change-record as 12,500,000 bytes.
+Thus, when this size-limit is reached, the oldest key-value pairs in the current record will be successively removed, by means of compaction.
+
+----
+curl -v -X POST http://localhost:8091/pools/default/buckets \
+-u Administrator:password \
+-d name=testBucket \
+-d ramQuota=1100 \
+-d storageBackend=magma \
+-d historyRetentionBytes=12500000
+----
+
+Success returns `202 Accepted`.
+
+[#example-flushenabled-edit]
+==== Example: Modify historyRetentionBytes, when Editing
+
+The following example modifies the value of `historyRetentionBytes` for the existing bucket `testBucket`.
+
+----
+curl -v -X POST http://localhost:8091/pools/default/buckets/testBucket \
+-u Administrator:password \
+-d historyRetentionBytes=12000000
+----
+
+Success returns `200 OK`.
+
+[#historyretentionseconds]
+=== historyRetentionSeconds
+
+Specifies the maximum number of seconds to be covered by the change history that is written to disk for each collection in the bucket (except the _&#95;default_ collection) when the value of `historyRetentionCollectionDefault` is `true`.
```

Comment:
The historyRetentionSeconds covers all data stored to disk, so all collections. Not just ones with history=true.

This is the same for the bytes configuration.

Basically these two configurations just tell magma that it must keep a "chunk" of storage which keeps all versions of documents that are flushed to disk.

If the user was writing data slowly to couchbase (1 update every 10 seconds for example) a complete change history will be retained, even without the collection history=true setting. However in this is no guarantee.

The collection configuration is a guarantee that no de-duplication will occur for writes to that collection, so they could do updates at 1000 per second or 1 per 10 second and all updates are preserved.

Reply:
Ah, I understand. So, if historyRetentionCollectionDefault is, say, true for the bucket, the 'history' setting per collection does not change that setting for the collection - it rather controls whether the change-history is done with or without de-duplication. Is that right?

Reply:
I've rephrased to express history as controlling deduplication, throughout.